extensions/jib/deployment/src/main/java/io/quarkus/jib/deployment/JibProcessor.java

```diff
@@ -0,0 +1,108 @@
+package io.quarkus.jib.deployment;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.jboss.logging.Logger;
+
+import com.google.cloud.tools.jib.api.Containerizer;
+import com.google.cloud.tools.jib.api.DockerDaemonImage;
+import com.google.cloud.tools.jib.api.ImageReference;
+import com.google.cloud.tools.jib.api.JavaContainerBuilder;
+import com.google.cloud.tools.jib.api.JibContainer;
+import com.google.cloud.tools.jib.api.JibContainerBuilder;
+
+import io.quarkus.bootstrap.util.ZipUtils;
+import io.quarkus.deployment.IsNormal;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationInfoBuildItem;
+import io.quarkus.deployment.pkg.builditem.ArtifactResultBuildItem;
+import io.quarkus.deployment.pkg.builditem.JarBuildItem;
+import io.quarkus.deployment.pkg.builditem.OutputTargetBuildItem;
+
+public class JibProcessor {
+
+    private static final Logger log = Logger.getLogger(JibProcessor.class);
+
+    private static final String MAIN_CLASS = "io.quarkus.runner.GeneratedMain";
```

Comment:
Should we consume `MainClassBuildItem` instead of hardcoding this value here? I am thinking of when the `cli` lands so it is rather future proof.

Reply:
That's probably a good idea, yeah!

Reply:
Done